src/main/java/seedu/contax/logic/parser/ParserUtil.java

```diff
@@ -238,4 +238,23 @@ public static Duration parseDuration(String duration) throws ParseException {
 
         return new Duration(convertedDuration);
     }
+
+    /**
+     * Parses a {@code String userCommand} into a {@code command with index}.
+     *
+     * @throws ParseException if the given {@code command} is empty.
+     */
+    public static String parseAndCreateNewCommand(String commandInput, String index) throws ParseException {
+        if (commandInput.trim().isEmpty()) {
+            throw new ParseException(Duration.MESSAGE_CONSTRAINTS);
+        }
+        StringBuilder output = new StringBuilder();
+        commandInput = commandInput.trim();
+        String[] splitCommand = commandInput.split(" ");
+        output.append(splitCommand[0]).append(" ").append(index).append(" ");
+        if (commandInput.contains(" ")) {
+            output.append(commandInput.substring(commandInput.indexOf(" ")));
```

Comment:
Why do you use `indexOf(" ")` instead of the length of `splitCommand[0]`?

Reply:
same effect, yah maybe better readablilty, changed